detect_secrets/plugins/common/util.py

```diff
@@ -34,7 +39,63 @@ def get_mapping_from_secret_type_to_class_name(custom_plugin_paths):
     }
 
 
-@_change_custom_plugin_paths_to_tuple
+def _dynamically_import_module(path_to_import, module_name):
+    """
+    :type path_to_import: str
+    :type module_name: str
+
+    :rtype: module
+    """
+    if is_python_2():  # pragma: no cover
+        return imp.load_source(
+            module_name,
+            path_to_import,
+        )
+
+    spec = importlib.util.spec_from_file_location(
+        module_name,
+        path_to_import,
+    )
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+def _is_valid_concrete_plugin_class(attr):
+    """
+    :type attr: Any
+
+    :rtype: bool
+    """
+    if (
+        not inspect.isclass(attr)
+        or
+        # Python 2 only:
+        # Old-style classes do not have the __mro__ attribute
+        not hasattr(attr, '__mro__')
+    ):
+        return False
+
+    # Dynamically imported classes have different
+    # addresses for the same functions as statically
+    # imported classes do, so issubclass does not work.
+    # We use __mro__ to loop through all parent classes.
+    # issubclass does work in Python 3, since parent classes
+    # do have the same addresses.
+    if not any(
+        'plugins.base.BasePlugin' in str(klass)
+        for klass in attr.__mro__
+    ):
+        return False
```

Comment:
Due to the issue explained in this comment, `super` calls in Python for plugins no longer work!

e.g. the `super` call in 
```python
    @property
    def __dict__(self):
        output = super(HighEntropyStringsPlugin, self).__dict__
        output.update({
            'hex_limit': self.entropy_limit,
        })

        return output
```
will throw a `super(type, obj): obj must be an instance or subtype of type` error because it thinks `self` is not a subtype, when really it just has different addresses for its functions. I patched around the same issue with `issubclass` in this highlighted code, but with `super` there is not a desirable path forward that I can see, I shall ask on StackOverflow for advice.

Reply:
For posterity, I did look into this, although I time-boxed myself.

You can change the metaclass of BasePlugin to be e.g.
```diff
+class AlwaysTrue(ABCMeta):
+    def __instancecheck__(self, instance):
+    def __subclasscheck__(self, instance):
```
however, super still throws the above error, only `isinstance` and `issubclass` work.

In other words, I believe this is definitely blocked on dropping Python 2 support https://github.com/Yelp/detect-secrets/issues/260. Either that or have the user supply the import path as well as the file path, but I don't want to do that.